### DISCUSSION TITLE
HP bar rework

### DISCUSSION THREAD

**Discussion by ghost** [diasurgical/DevilutionX#5840] [Ideas]
As i mentioned in the previous discussion thread i'd love to see a rework of the HP bar. I've made a rough draft of my idea.

![HP Bar rework](https://user-images.githubusercontent.com/53155671/221575851-25f50bc8-dffd-4ab4-9dde-1d490cb8c9c3.png)

It would take me quite a bit of time and effort to get it to look as awesome as the original. I honestly think the original artist did a much better job than i can do, but i'm willing to put in some effort if it has interest.

I made this image in inkscape which is what i usually use. I can share the SVG if anyone is interested.

Edit: I considered putting in an image of the class that did damage instead of the "Drop of blood" idea, but i think it will take up too much space. The current in-game size of the HP bar is great and i dont want to bloat it.

Edit: My "final" take on a re-work:
![g3102](https://user-images.githubusercontent.com/53155671/221780550-d517df38-9912-412e-9a2a-f1ba2f22cf45.png)

[I covered it in a comment further down.](https://github.com/diasurgical/devilutionX/discussions/5840#discussioncomment-5138940)

**Comment by AJenbo** [comment]
Another option could be:
![image](https://user-images.githubusercontent.com/204594/221581640-aca05123-3a3d-4375-b3ee-8714011511c0.png)

This by the way was my original proposal:
![image](https://user-images.githubusercontent.com/204594/221580816-82813cb4-6371-4a5a-bc35-849dcc40a42e.png)

**Reply by AJenbo** [reply]
The fill color is the same as on for the player on the map

**Reply by ghost** [reply]
I like the bottom one the most, but i think it would be better to have everything below the bar. 

It's a matter of directing peoples eyes to the relevant information. Thats why i put everything below the bar.

Also you still don't have the monster type tag i crave ;)

**Reply by AJenbo** [reply]
I didn't have any suggestions for that part

**Reply by DakkJaniels** [reply]
I like the 2nd one the most.  I think having things compact is better than having everything below the bar, so just looking to the left gives you most of the information you need. One thing that might be useful though is if the box representing _your player_ is denoted differently, so it's clear (at least to the person that needs to take action) who needs to hit the monster still. Since it's not immediately clear what the player order is, that might be the most useful.

If we are talking about changing things, maybe little icons for undead, demon, and animal could go where the blood drop is in that first picture?

**Reply by ghost** [reply]
I also like the compactness of the 2nd picture in AJenbo's post. The drop of blood idea was based on making it more intuitive to see that it was about damage.

An icon could be cool. Skull for undead, Horned head for demon and....idk for animal :)

**Reply by AJenbo** [reply]
I think a rat is the norm

**Reply by AJenbo** [reply]
or a paw

**Reply by ghost** [reply]
It's diablo. Maybe a tiger claw? Needs to fit with the dark fantasy setting.

Edit: A wolf! Or Raven! Western dark fantasy setting. We already have druids summoning wolves and ravens in Diablo 2.

**Reply by DakkJaniels** [reply]
a 28x28 px claw already exists in the game, which might work (not sure if that size is too big for this though): 
![image](https://user-images.githubusercontent.com/6080734/221592003-e512eebe-5839-49f8-9dd8-b64c5c208078.png)

Not sure if other graphics already exist for undead and demon.

**Reply by ghost** [reply]
I think the main problem is just selecting which type of icons would be best. Skulls, Wolf-heads and horned heads are so common that getting an image shouldn't be a problem. I actually drew the shield and blood drop myself in 15-20 min tops because tutorials where readily available. I think the same would apply to other simple well-known icons.

**Comment by ghost** [comment]
Alright, so i did three more takes on a draft.

![HP Bars](https://user-images.githubusercontent.com/53155671/221642428-9ef966ce-d318-4d91-9a37-02f18a417689.png)

Have not solved where the "Drop of blood" should be. I considered putting them top-left inside the hp-bar, but then they cannot be filled with red for obvious reasons. I also think they could be made a bit smaller.

**Reply by ghost** [reply]
![HP Bar](https://user-images.githubusercontent.com/53155671/221647772-de6d5310-f8cc-4a42-8d58-6870083f30a1.png)

Another take

**Comment by Trihedraf** [comment]
I don't think we should move to all red indicators for hit on monster, because the current system you can tell by looking at the map which color is which player.

**Comment by ghost** [comment]
Alright, so i slept on it and i think i've made a draft that everyone will be able to agree on and be happy with! :D

![g3102](https://user-images.githubusercontent.com/53155671/221779267-28e81de9-4d3d-445c-95b6-4136a5297fb1.png)


Instead of squares and drops of blood i have crossed swords - An indicator that you are engaged with the monster. The swords should be colored as they are in 1.4.1.

I also moved the symbols to the original position. It slightly bugs me that its above the bar, but i think you guys are right. It is better to have all the information on the left side.

I think a small tag is better than icons for indicating monster type.


**Edit:** You could actually have Sword for warrior, Staff for sorcerer and Bow for rogue.

**Edit 2:** Is it possible to only have the icons show up when 2 or more players are in the game? It is useless information if you're soloing anyway.

**Reply by AJenbo** [reply]
Does this work?
![image](https://user-images.githubusercontent.com/204594/221791800-8c77665d-b059-4017-8488-1fcd85218f99.png)

**Reply by ghost** [reply]
I thought about it, but concluded that it might obscure part of the hp bar.

What do you think of the rest of it?

**Reply by ghost** [reply]
I arrive at this if i keep working with the icon idea.

![g2592](https://user-images.githubusercontent.com/53155671/221836199-14862165-6851-4196-8332-60691023e8d5.png)

The hp bar is a little bit shorter, but that shouldn't be a problem.

The icons should be white for common monster, blue for minion and gold-unique for boss monster.
